presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergResourceFactory.java

```diff
@@ -134,6 +144,29 @@ public Map<String, String> getCatalogProperties(ConnectorSession session)
         if (catalogWarehouse != null) {
             properties.put(WAREHOUSE_LOCATION, catalogWarehouse);
         }
+        if (catalogType == NESSIE) {
+            properties.put("ref", IcebergSessionProperties.getNessieReferenceName(session));
+            properties.put("uri", nessieConfig.getServerUri().orElseThrow(() -> new IllegalStateException("iceberg.nessie.uri must be set for Nessie")));
+            String hash = IcebergSessionProperties.getNessieReferenceHash(session);
+            if (hash != null) {
+                properties.put("ref.hash", hash);
+            }
+            nessieConfig.getReadTimeoutMillis().ifPresent(val -> properties.put("transport.read-timeout", val.toString()));
+            nessieConfig.getConnectTimeoutMillis().ifPresent(val -> properties.put("transport.connect-timeout", val.toString()));
+            nessieConfig.getClientBuilderImpl().ifPresent(val -> properties.put("client-builder-impl", val));
+            nessieConfig.getAuthenticationType().ifPresent(type -> {
+                if (type.equals(BASIC) && nessieConfig.getUsername().isPresent() && nessieConfig.getPassword().isPresent()) {
+                    properties.put("authentication.username", nessieConfig.getUsername().get());
+                    properties.put("authentication.password", nessieConfig.getPassword().get());
+                }
+                else if (type.equals(BEARER) && nessieConfig.getBearerToken().isPresent()) {
+                    properties.put("authentication.token", nessieConfig.getBearerToken().get());
+                }
+            });
+            if (!nessieConfig.isCompressionEnabled()) {
```

Comment:
nit: why not make this config as isCompressionDisabled?

Reply:
makes sense, fixed